pyuvdata/miriad.py

```diff
@@ -396,18 +398,50 @@ def read_miriad(self, filepath, correct_lat_lon=True, run_check=True, run_check_
                 ra_list[blt_index] = ra_pol_list[blt_index, 0]
                 dec_list[blt_index] = dec_pol_list[blt_index, 0]
 
-        # check if ra is constant throughout file; if it is,
-        # file is tracking if not, file is drift scanning
-        blt_good = np.where(~np.all(self.flag_array, axis=(1, 2, 3)))
-        if np.isclose(np.mean(np.diff(ra_list[blt_good])), 0.):
-            self.set_phased()
+        # first check to see if the phase_type was specified.
+        if phase_type is not None:
+            if phase_type is 'phased':
+                self.set_phased()
+            elif phase_type is 'drift':
+                self.set_drift()
+            else:
+                raise ValueError('The phase_type was not recognized. '
+                                 'Set the phase_type to "drift" or "phased" to '
+                                 'reflect the phasing status of the data')
+        else:
+            # check if ra is constant throughout file; if it is,
+            # file is tracking if not, file is drift scanning
+            if self.Ntimes > 1:
+                blt_good = np.where(~np.all(self.flag_array, axis=(1, 2, 3)))
+                if np.isclose(np.mean(np.diff(ra_list[blt_good])), 0.):
+                    self.set_phased()
+                else:
+                    self.set_drift()
+            else:
+                # if there's only one time, checking for consistent RAs doesn't work.
+                # instead check for the presence of an epoch variable, which isn't
+                # really a good option, but at least it prevents crashes.
+                if 'epoch' in uv.vartable.keys():
```

Comment:
Technically if there is only one time, tracking and drift are indistinguishable. However, pyuvdata does assume that "drift" implies drifting while phased at zenith. So perhaps a better check would be to see if the phase center is set to zenith? Otherwise we could be messing up files.